libraries/helpers.rb

```diff
@@ -74,5 +76,19 @@ def home_basedir
         '/home'
       end
     end
+
+    # Returns the type of the public key or nil
+    #
+    # @return [String]
+    def pubkey_type(pubkey)
```

Comment:
This can be simplified to something like:
```ruby
def pubkey_type(pubkey)
  %w(ed25519 dss rsa dsa ecdsa).filter { |kt| kt.eql?(pubkey.split.first) }.first || 'rsa'
end
```
If nothing matches `filter` will return an empty array which `.first` on it will return `nil`, so it makes it a nice one-liner.

Reply:
Almost worked the first time.

Had to reorder the array to make sure dsa and ecdsa both work and use include instead of eql :) But quite a nice one liner indeed.